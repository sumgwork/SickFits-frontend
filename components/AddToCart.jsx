import React from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";
import PropTypes from "prop-types";

const ADD_TO_CART_MUTATION = gql`
  mutation ADD_TO_CART_MUTATION($id: ID!) {
    addToCart(id: $id) {
      id
      count
    }
  }
`;

const AddToCart = props => {
  return (
    <Mutation mutation={ADD_TO_CART_MUTATION} variables={{ id: props.item.id }}>
      {addToCart => <button onClick={addToCart}>Add to cart ðŸ›’</button>}
    </Mutation>
  );
};

AddToCart.propTypes = {
  item: PropTypes.object.isRequired
};

export default AddToCart;
