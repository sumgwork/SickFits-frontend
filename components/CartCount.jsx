import React, { Component } from "react";
import PropTypes from "prop-types";
import { TransitionGroup, CSSTransition } from "react-transition-group";
import styled from "styled-components";

const Dot = styled.div`
  background: ${props => props.theme.red};
  color: white;
  border-radius: 50%;
  padding: 0.5rem;
  line-height: 2rem;
  min-width: 3rem;
  margin-left: 1rem;
  font-weight: 600;
  font-feature-settings: "tnum";
  font-variant-numeric: tabular-nums;
  /* height: 3rem;
  text-align: center; */
`;

const AnimationStyles = styled.span`
  position: relative;
  .count {
    display: block;
    position: relative;
    transition: all 0.4s;
    backface-visibility: hidden;
  }
  .count-enter {
    transform: rotateX(0.5turn);
  }
  .count-enter-active {
    transform: rotateX(0);
  }

  .count-exit {
    transform: rotateX(0);
    top: 0;
    position: absolute;
  }
  .count-exit-active {
    transform: rotateX(0.5turn);
  }
`;

class CartCount extends Component {
  state = {};

  render() {
    const { count } = this.props;
    return (
      <AnimationStyles>
        <TransitionGroup>
          <CSSTransition
            className="count"
            classNames="count"
            key={count}
            timeout={{ enter: 400, exit: 400 }}
            unmountOnExit
          >
            <Dot>{count}</Dot>
          </CSSTransition>
        </TransitionGroup>
      </AnimationStyles>
    );
  }
}

export default CartCount;
