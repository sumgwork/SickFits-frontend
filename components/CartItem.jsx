import React from "react";
import styled from "styled-components";
import formatMoney from "../lib/formatMoney";

const CartItemStyles = styled.li`
  padding: 1rem 0;
  border-bottom: 1px solid ${props => props.theme.lightgrey};
  display: grid;
  grid-template-columns: auto 1fr auto;
  align-items: center;
  img {
    margin-right: 10px;
  }
  h3,
  p {
    margin: 0;
  }
`;

const CartItem = ({ item, count }) => {
  return (
    <CartItemStyles>
      <img width="100" src={item.image} alt={item.title} />
      <div className="cart-item-details">
        <h3>{item.title}</h3>
        <p>
          {formatMoney(item.price * count)}
          {" - "}
          <em>
            {count} &times; {formatMoney(item.price)}
          </em>
        </p>
      </div>
    </CartItemStyles>
  );
};

export default CartItem;
