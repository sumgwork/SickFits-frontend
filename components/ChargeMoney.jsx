import React, { Component } from "react";
import StripeCheckout from "react-stripe-checkout";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";
import Router from "next/router";
import User from "./User";
import calcTotalPrice from "../lib/calcTotalPrice";

function totalItems(cart) {
  return cart.reduce((tally, cartItem) => tally + cartItem.count, 0);
}

class ChargeMoney extends Component {
  state = {};

  onToken(res) {
    console.log("token", res);
  }
  render() {
    return (
      <User>
        {({ data: { me } }) => (
          <StripeCheckout
            amount={calcTotalPrice(me.cart)}
            name="SG Shop"
            description={`Order of ${totalItems(me.cart)} items`}
            image={me.cart[0].item && me.cart[0].item.image}
            stripeKey="pk_test_mNJ7LmiQETKJwrhsJrq2Iskx"
            currency="USD"
            email={me.email}
            token={res => this.onToken(res)}
          >
            {this.props.children}
          </StripeCheckout>
        )}
      </User>
    );
  }
}

export default ChargeMoney;
