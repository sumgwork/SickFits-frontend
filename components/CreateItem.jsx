import React, { Component } from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";
import styled from "styled-components";
import Router from "next/router";
import formatMoney from "../lib/formatMoney";
import Form from "./styles/Form";
import Error from "./ErrorMessage";

const CLOUDINARY_URL = "https://api.cloudinary.com/v1_1/sumitgovilstudy";
const CLOUDINARY_UPLOAD_PRESET = "sickfits";

const CREATE_ITEM_MUTATION = gql`
  mutation CREATE_ITEM_MUTATION(
    $title: String!
    $description: String!
    $price: Int!
    $image: String
    $largeImage: String
  ) {
    createItem(
      title: $title
      description: $description
      price: $price
      image: $image
      largeImage: $largeImage
    ) {
      id
    }
  }
`;

const ImageHolder = styled.div`
  text-align: center;
`;

class CreateItem extends Component {
  state = {
    title: "",
    description: "",
    price: 0,
    image: "",
    largeImage: ""
  };

  uploadFile = async e => {
    const files = e.target.files;
    const data = new FormData();
    data.append("file", files[0]);
    data.append("upload_preset", CLOUDINARY_UPLOAD_PRESET);
    const res = await fetch(`${CLOUDINARY_URL}/upload`, {
      method: "POST",
      body: data
    });

    const file = await res.json();
    this.setState({
      image: file.secure_url,
      largeImage: file.eager[0].secure_url
    });
  };

  handleChange = e => {
    const { name, type, value } = e.target;
    const val = type === "number" ? parseFloat(value) : value;
    this.setState({
      [name]: val
    });
  };

  render() {
    return (
      <div>
        <Mutation mutation={CREATE_ITEM_MUTATION} variables={this.state}>
          {(createItem, { loading, error, called, data }) => (
            <Form
              onSubmit={async e => {
                e.preventDefault();
                const res = await createItem();
                Router.push({
                  pathname: "/item",
                  query: { id: res.data.createItem.id }
                });
              }}
              data-test="form"
            >
              <Error error={error} />
              <fieldset aria-busy={loading} disabled={loading}>
                <ImageHolder>
                  {this.state.image && (
                    <img src={this.state.image} alt="image" />
                  )}
                </ImageHolder>

                <label htmlFor="file">
                  Image
                  <input
                    type="file"
                    id="file"
                    placeholder="Upload file"
                    name="file"
                    onChange={this.uploadFile}
                    required
                  />
                </label>

                <label htmlFor="title">
                  Title
                  <input
                    type="text"
                    id="title"
                    placeholder="Title"
                    name="title"
                    value={this.state.title}
                    onChange={this.handleChange}
                    required
                  />
                </label>

                <label htmlFor="price">
                  Price
                  <input
                    type="number"
                    id="price"
                    placeholder="Price"
                    name="price"
                    value={this.state.price}
                    onChange={this.handleChange}
                    required
                  />
                </label>

                <label htmlFor="description">
                  Description
                  <textarea
                    id="description"
                    placeholder="Enter description of the item"
                    name="description"
                    value={this.state.description}
                    onChange={this.handleChange}
                    required
                  />
                </label>

                <button type="submit">Submit</button>
              </fieldset>
            </Form>
          )}
        </Mutation>
      </div>
    );
  }
}

export default CreateItem;
export { CREATE_ITEM_MUTATION };
