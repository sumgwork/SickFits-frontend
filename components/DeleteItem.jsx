import React, { Component } from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";

import Error from "./ErrorMessage";
import { GET_ALL_ITEMS } from "./Items";

const DELETE_ITEM = gql`
  mutation DELETE_ITEM($id: ID!) {
    deleteItem(id: $id) {
      id
    }
  }
`;

class DeleteItem extends Component {
  update(cache, payload) {
    const data = cache.readQuery({ query: GET_ALL_ITEMS });
    data.items = data.items.filter(
      item => item.id !== payload.data.deleteItem.id
    );
    cache.writeQuery({ query: GET_ALL_ITEMS, data });
  }
  render() {
    return (
      <Mutation
        mutation={DELETE_ITEM}
        variables={{ id: this.props.id }}
        update={this.update}
      >
        {(deleteItem, { error, loading }) => {
          // if (error) return <Error />;
          if (loading) return <p>loading</p>;
          return (
            <button
              onClick={() => {
                if (confirm("Are you sure you want to delete this item?")) {
                  deleteItem().catch(e => {
                    alert(e);
                  });
                }
              }}
            >
              {this.props.children}
            </button>
          );
        }}
      </Mutation>
    );
  }
}

export default DeleteItem;
