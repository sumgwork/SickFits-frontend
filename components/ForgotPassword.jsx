import React, { Component } from "react";
import { Mutation } from "react-apollo";
import gql from "graphql-tag";
import Form from "./styles/Form";
import Error from "./ErrorMessage";

const REQUEST_RESET_MUTATION = gql`
  mutation REQUEST_RESET_MUTATION($email: String!) {
    requestReset(email: $email) {
      message
    }
  }
`;

class ForgotPassword extends Component {
  state = { email: "" };

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({
      [name]: value
    });
  };
  render() {
    return (
      <Mutation mutation={REQUEST_RESET_MUTATION} variables={this.state}>
        {(requestReset, { error, loading, called }) => {
          return (
            <Form
              disabled={loading}
              aria-busy={loading}
              method="post"
              onSubmit={async e => {
                e.preventDefault();
                await requestReset();
                this.setState({ email: "" });
              }}
            >
              <h2>Forgot Password</h2>
              {loading && <h1>Loading</h1>}
              {!loading && !error && called && <p>Check your email!</p>}
              <Error error={error} />
              <fieldset>
                <label htmlFor="email">
                  Email
                  <input
                    type="email"
                    placeholder="Email"
                    name="email"
                    value={this.state.email}
                    onChange={this.handleChange}
                    required
                  />
                </label>
              </fieldset>
              <button type="submit">Request Reset</button>
            </Form>
          );
        }}
      </Mutation>
    );
  }
}

export default ForgotPassword;
