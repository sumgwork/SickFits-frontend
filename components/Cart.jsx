import React from "react";
import { Query, Mutation } from "react-apollo";
import gql from "graphql-tag";

import CartStyles from "./styles/CartStyles";
import Supreme from "./styles/Supreme";
import CloseButton from "./styles/CloseButton";
import ShopButton from "./styles/ShopButton";
import User from "./User";
import CartItem from "./CartItem";
import calcTotalPrice from "../lib/calcTotalPrice";
import formatMoney from "../lib/formatMoney";

const LOCAL_STATE_QUERY = gql`
  query {
    cartOpen @client
  }
`;

const TOGGLE_CART_MUTATION = gql`
  mutation {
    toggleCart @client
  }
`;

const Cart = props => {
  return (
    <User>
      {({ data: { me } }) => {
        if (!me) return null;
        return (
          <Mutation mutation={TOGGLE_CART_MUTATION}>
            {toggleCart => (
              <Query query={LOCAL_STATE_QUERY}>
                {({ data }) => (
                  <CartStyles open={data && data.cartOpen}>
                    <header>
                      <CloseButton title="close" onClick={toggleCart}>
                        &times;
                      </CloseButton>
                      <Supreme>{me.name}'s Cart</Supreme>
                      <p>
                        You have {me.cart.length} item
                        {me.cart.length <= 1 ? "" : "s"} in your cart.
                      </p>
                    </header>
                    <main>
                      <ul>
                        {me.cart.map(cartItem => (
                          <CartItem
                            key={cartItem.id}
                            item={cartItem.item}
                            count={cartItem.count}
                          />
                        ))}
                      </ul>
                    </main>
                    <footer>
                      <p>{formatMoney(calcTotalPrice(me.cart))}</p>
                      <ShopButton>Checkout</ShopButton>
                    </footer>
                  </CartStyles>
                )}
              </Query>
            )}
          </Mutation>
        );
      }}
    </User>
  );
};

export default Cart;
export { LOCAL_STATE_QUERY, TOGGLE_CART_MUTATION };
