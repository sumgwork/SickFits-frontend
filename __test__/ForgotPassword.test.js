import { mount } from "enzyme";
import wait from "waait";
import toJSON from "enzyme-to-json";
import ForgotPassword from "../components/ForgotPassword";
import { REQUEST_RESET_MUTATION } from "../components/ForgotPassword";
import { MockedProvider } from "react-apollo/test-utils";

const mocks = [
  {
    request: {
      query: REQUEST_RESET_MUTATION,
      variables: { email: "sumit.govil@optus.com.au" }
    },
    result: {
      data: {
        requestReset: {
          message: "success",
          __typename: "message"
        }
      }
    }
  }
];

describe("<ForgotPassword />", () => {
  it("renders and matches snapshot", async () => {
    const wrapper = mount(
      <MockedProvider>
        <ForgotPassword />
      </MockedProvider>
    );
    const form = wrapper.find('form[data-test="form"]');
    console.log(form.debug());
    expect(toJSON(form)).toMatchSnapshot();
  });
});
