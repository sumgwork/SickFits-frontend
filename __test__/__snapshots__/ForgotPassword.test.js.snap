// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPassword /> renders and matches snapshot 1`] = `
<form
  aria-busy={false}
  className="Form-sc-1xszr8q-0 leWalr"
  data-test="form"
  disabled={false}
  method="post"
  onSubmit={[Function]}
>
  <h2>
    Forgot Password
  </h2>
  <DisplayError
    error={Object {}}
  />
  <fieldset>
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
    </label>
  </fieldset>
  <button
    type="submit"
  >
    Request Reset
  </button>
</form>
`;
